core/src/main/java/hudson/tasks/Maven.java

```diff
@@ -340,11 +340,12 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListene
                 }
             }
 
+            Set<String> sensitiveVars = build.getSensitiveBuildVariables();
+            final VariableResolver<String> resolver = new Union<String>(new ByMap<String>(env), vr);
+            args.addKeyValuePairsFromPropertyString("-D", this.properties, resolver, sensitiveVars);
```

Comment:
An issue that's been around for ~5 months in weeklies and 2 months in LTS (since it was present in the baseline) warrants a special exception usually reserved for critical issues? Where are the hordes of users complaining about this? What am I missing?